core/frontend/src/store/bridget.ts

```diff
@@ -43,9 +53,91 @@ class BridgetStore extends VuexModule {
     this.available_serial_ports = available_serial_ports
     this.updating_bridges = false
   }
+
+  @Mutation
+  registerListener(): void {
+    this.listener_number += 1
+  }
+
+  @Mutation
+  deregisterListener(): void {
+    if (this.listener_number <= 0) {
+      console.error('Deregister called with zero listeners.')
+      return
+    }
+    this.listener_number -= 1
+  }
+
+  @Action
+  async fetchAvailableBridges(): Promise<void> {
+    if (prefetched_bridges && !this.listener_number) {
+      return
+    }
+    back_axios({
+      method: 'get',
+      url: `${this.API_URL}/bridges`,
+      timeout: 10000,
+    })
+      .then((response) => {
+        const available_bridges = response.data
+        this.setAvailableBridges(available_bridges)
+        prefetched_bridges = true
+      })
+      .catch((error) => {
+        this.setAvailableBridges([])
+        if (error === backend_offline_error) { return }
+        const message = `Could not fetch available bridges: ${error.message}`
+        notifications.pushError({ service: bridget_service, type: 'BRIDGES_FETCH_FAIL', message })
+      })
+      .finally(() => {
+        this.setUpdatingBridges(false)
+      })
+  }
+
+  @Action
+  async fetchAvailableSerialPorts(): Promise<void> {
+    if (prefetched_serial && !this.listener_number) {
+      return
+    }
+    back_axios({
+      method: 'get',
+      url: `${this.API_URL}/serial_ports`,
+      timeout: 10000,
+    })
+      .then((response) => {
+        const available_ports = response.data
+        this.setAvailableSerialPorts(available_ports)
+        prefetched_serial = true
+      })
+      .catch((error) => {
+        this.setAvailableSerialPorts([])
+        if (error === backend_offline_error) { return }
+        const message = `Could not fetch available serial ports: ${error.message}`
+        notifications.pushError({ service: bridget_service, type: 'BRIDGET_SERIAL_PORTS_FETCH_FAIL', message })
+      })
+      .finally(() => {
+        this.setUpdatingSerialPorts(false)
+      })
+  }
+
+  @Action
+  async registerObject(object: any): Promise<void> {
+    this.registerListener()
+    const ref = new WeakRef(object)
+    const id = setInterval(() => {
+      // Check if object does not exist anymore or if it was destroyed by vue
+      // eslint-disable-next-line
+      if (ref.deref()?._isDestroyed) {
+        this.deregisterListener()
+        clearInterval(id)
+      }
+    }, 1000)
+  }
```

Comment:
Can we somehow move the register/deregister/listener/object logic to a separate file, to be commonly used by all stores?